edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesModule.cs

```diff
@@ -84,8 +99,10 @@ public bool Equals(KubernetesModule<TConfig> other)
                 this.RestartPolicy == other.RestartPolicy &&
                 this.ConfigurationInfo == other.ConfigurationInfo &&
                 this.ImagePullPolicy == other.ImagePullPolicy &&
-                this.Config is DockerConfig &&
-                this.Config as DockerConfig == other.Config as DockerConfig &&
+                this.Config is CombinedDockerConfig &&
+                this.Config as CombinedDockerConfig == other.Config as CombinedDockerConfig &&
+                this.Config is string &&
+                this.Config as string == other.Config as string &&
```

Comment:
This is a really strange set of boolean conditions...

Reply:
Yes, let me check that more carefully - It's probably not doing what I expect.

Reply:
I agree it looks incorrect